### PR TITLE
Integrate change log api with the frontend

### LINKED ISSUE #643: [Feature] Integrate change log api with the frontend
**What is the problem?**
[Change log service](https://github.com/short-d/short/blob/master/frontend/src/service/ChangeLog.service.ts) in the frontend is rendering dummy data hard-coded. 

**Your solution**
Integrate change log service with [GraphQL API](https://github.com/short-d/short/blob/033ccc686046e12f72cdfdbdd3b829bebfab88e1/backend/app/adapter/graphql/schema.go#L17) so that change log modal works as a whole.

### PR DESCRIPTION
**Pull request by kapillamba4**
Fixes #643

### CODE REVIEW

File: frontend/src/component/pages/HomePage.tsx [hunk 1]
```diff
@@ -391,6 +395,7 @@ export class HomePage extends Component<Props, State> {
         <ChangeLogModal
           ref={this.changeLogModalRef}
           changeLog={this.state.changeLog}
+          onClose={this.handleChangeLogClose}
```
Comment (byliuyang) [inline frontend/src/component/pages/HomePage.tsx:398]: Not on close. What happens if the user directly closes the web browser? Let's invoke `viewChangeLog` when the modal is opened.

File: frontend/src/service/ChangeLogGraphQL.api.ts [hunk 2]
```diff
@@ -71,6 +81,45 @@ export class ChangeLogGraphQLApi {
...
+
+      const variables = {
+        authToken: this.authService.getAuthToken(),
+        captchaResponse: captchaResponse
+      };
+
+      try {
+        const res: IChangeLogGraphQLMutation = await this.graphQLService.mutate<
+          IChangeLogGraphQLMutation
+        >(this.baseURL, {
```
Comment (byliuyang) [inline frontend/src/service/ChangeLogGraphQL.api.ts:109]:
```suggestion
        const res: IChangeLogGraphQLMutation = await this.graphQLService.mutate<IChangeLogGraphQLMutation>
        (this.baseURL, {
```

Comment (kapillamba4) [reply frontend/src/service/ChangeLogGraphQL.api.ts:109]: This is due to ./script/format

File: frontend/src/component/pages/HomePage.tsx [hunk 3]
```diff
@@ -305,6 +305,8 @@ export class HomePage extends Component<Props, State> {
     if (this.changeLogModalRef.current) {
       this.changeLogModalRef.current.open();
     }
+
+    this.props.changeLogService.viewChangeLog();
```
Comment (byliuyang) [inline frontend/src/component/pages/HomePage.tsx:309]: Sorry, I mean handle `onOpenEvent` for change log modal.

File: frontend/src/component/ui/ChangeLogModal.tsx [hunk 4]
```diff
@@ -14,6 +14,7 @@ interface State {
 interface Props {
   changeLog?: Array<Change>;
   defaultVisibleLogs: number;
+  onOpen?: () => void;
```
Comment (byliuyang) [inline frontend/src/component/ui/ChangeLogModal.tsx:17]:
```suggestion
  onModalOpen?: () => void;
```

File: frontend/src/component/ui/ChangeLogModal.tsx [hunk 5]
```diff
@@ -75,7 +76,11 @@ export class ChangeLogModal extends Component<Props, State> {
 
   render() {
     return (
-      <Modal ref={this.modalRef} canClose={true}>
+      <Modal
+        ref={this.modalRef}
+        canClose={true}
+        onModalOpen={this.props.onOpen}
```
Comment (byliuyang) [inline frontend/src/component/ui/ChangeLogModal.tsx:82]:
```suggestion
        onModalOpen={this.props.onModalOpen}
```

File: frontend/src/component/pages/HomePage.tsx [hunk 6]
```diff
@@ -390,6 +394,7 @@ export class HomePage extends Component<Props, State> {
         />
         <ChangeLogModal
           ref={this.changeLogModalRef}
+          onOpen={this.handleOpenChangeLogModal}
```
Comment (byliuyang) [inline frontend/src/component/pages/HomePage.tsx:397]:
```suggestion
          onModalOpen={this.handleOpenChangeLogModal}
```

**Pull request merged**